rest/connector-generator/src/main/java/io/syndesis/connector/generator/swagger/SwaggerConnectorGenerator.java

```diff
@@ -76,24 +78,49 @@
     }
 
     @Override
-    public Connector generate(final ConnectorTemplate connectorTemplate, final Connector template) {
-        final Map<String, String> configuredProperties = template.getConfiguredProperties();
+    public Connector generate(final ConnectorTemplate connectorTemplate, final CustomConnector customConnector) {
+        final Connector connector = basicConnector(connectorTemplate, customConnector);
```

Comment:
So a `Connector` is created from a `CustomConnector` ? From the naming I first implied that a CustomConnector **is** a Connector (with some kind of specialization). Isn't it more like a `ApiConnectorTemplate` or `ApiConnectorSource` ?

Reply:
Please correct me, but if I understand right, then a 'ConnectorTemplate' has the type information and the  'CustomConnector'  the runtime information ?

Reply:
Yeah, `ConnectorTemplate` was already taken. Here is the lowdown:
 - `ConnectorTemplate` should define all non user provided aspects of a new connector
 - `CustomConnector` defines only user provided aspects
 - `ConnectorGenerator` mixes those two to produce a `Connector`

Given that the `ConnectorTemplate` and `Connector` are taken I could not find a better name than a `CustomConnector`. Naming is hard.

You might also notice that this goes beyond API connectors, and allows creating any kind of connector based on the template and generator implementation.

Reply:
What's about 'ConnectorParameters' instead of 'CustomConnector' ? 

I mean, you have a _connector template_ fill in some _connector parameters_ and then get a _connector_ . Sounds like a common 'template pattern' ...

Even if it is not 100% fit (as a connector takes a part from the template, and another part from the customconnector) it has at least a known semantics. Does this catch your case, too ?